lib/srv/regular/proxy.go

```diff
@@ -260,12 +263,46 @@ func (t *proxySubsys) proxyToHost(ctx context.Context, ch ssh.Channel, clientSrc
 	}
 
 	go func() {
-		t.close(utils.ProxyConn(ctx, ch, conn))
+		t.close(utils.ProxyConn(ctx, &checkedPrefixReader{
```

Comment:
The correct place for this is `(*Router).DialHost` in `lib/proxy/router.go`, which ends up being used by proxy subsys and TransportService/ProxySSH (and even then, maybe only in places where we end up actually dialing a connection to the outside world?).